snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ecl/SnomedEclRefinementEvaluator.java

```diff
@@ -487,6 +495,169 @@ public SnomedEclRefinementEvaluator(EclExpression focusConcepts) {
+			value = new RelationshipValue(((IntegerValueNotEquals) comparison).getValue());
+			operator = SearchResourceRequest.Operator.NOT_EQUALS;
+		} else if (comparison instanceof DecimalValueEquals) {
+			value = new RelationshipValue(((DecimalValueEquals) comparison).getValue().doubleValue());
+			operator = SearchResourceRequest.Operator.EQUALS;
+		} else if (comparison instanceof DecimalValueNotEquals) {
+			value = new RelationshipValue(((DecimalValueNotEquals) comparison).getValue().doubleValue());
+			operator = SearchResourceRequest.Operator.NOT_EQUALS;
+		} else if (comparison instanceof IntegerValueLessThan) {
+			value = new RelationshipValue(((IntegerValueLessThan) comparison).getValue());
+			operator = SearchResourceRequest.Operator.LESS_THAN;
+		} else if (comparison instanceof DecimalValueLessThan) {
+			value = new RelationshipValue(((DecimalValueLessThan) comparison).getValue().doubleValue());
+			operator = SearchResourceRequest.Operator.LESS_THAN;
+		} else if (comparison instanceof IntegerValueLessThanEquals) {
+			value = new RelationshipValue(((IntegerValueLessThanEquals) comparison).getValue());
+			operator = SearchResourceRequest.Operator.LESS_THAN_EQUALS;
+		} else if (comparison instanceof DecimalValueLessThanEquals) {
+			value = new RelationshipValue(((DecimalValueLessThanEquals) comparison).getValue().doubleValue());
+			operator = SearchResourceRequest.Operator.LESS_THAN_EQUALS;
+		} else if (comparison instanceof IntegerValueGreaterThan) {
+			value = new RelationshipValue(((IntegerValueGreaterThan) comparison).getValue());
+			operator = SearchResourceRequest.Operator.GREATER_THAN;
+		} else if (comparison instanceof DecimalValueGreaterThan) {
+			value = new RelationshipValue(((DecimalValueGreaterThan) comparison).getValue().doubleValue());
+			operator = SearchResourceRequest.Operator.GREATER_THAN;
+		} else if (comparison instanceof IntegerValueGreaterThanEquals) {
+			value = new RelationshipValue(((IntegerValueGreaterThanEquals) comparison).getValue());
+			operator = SearchResourceRequest.Operator.GREATER_THAN_EQUALS;
+		} else if (comparison instanceof DecimalValueGreaterThanEquals) {
+			value = new RelationshipValue(((DecimalValueGreaterThanEquals) comparison).getValue().doubleValue());
+			operator = SearchResourceRequest.Operator.GREATER_THAN_EQUALS;
+		} else {
+			return SnomedEclEvaluationRequest.throwUnsupported(comparison);
+		}
+		
+		return evalStatementsWithValue(context, focusConceptIds, typeIds, value, operator);
+	}
+
+	private Promise<Collection<Property>> evalStatementsWithValue(
+		final BranchContext context, 
+		final Set<String> focusConceptIds,
+		final Collection<String> typeIds, 
+		final RelationshipValue value, 
+		final SearchResourceRequest.Operator operator) {
+
+		// TODO: does this request need to support filtering by group?
+		Promise<Collection<Property>> relationships = SnomedRequests.prepareSearchRelationship()
+				.all()
```

Comment:
To reduce memory-pressure and improve speed, use field selection here to reduce the number of fields loaded from the concrete value relationship. Same as loading fields during standard relationship search.